internal/registration/registration.go

```diff
@@ -294,17 +311,16 @@ func (r *Registration) Deregister() error {
 		}
 	}
 
+	r.lock.RLock()
```

Comment:
Is that the only fix for the race condition or are there other changes that are part of the fix?

Reply:
nope, that's the only fix

Reply:
:+1: